src/main/java/net/tiagofar78/prisonescape/game/PrisonEscapeGame.java

```diff
@@ -826,6 +846,28 @@ public void policeInspectedPrisioner(String policeName, String prisionerName) {
         }
     }
 
+    public void playerInteractWithDoor(
+            PrisonEscapePlayer player,
+            int inventoryIndex,
+            Item itemHeld,
+            Door door,
+            PrisonEscapeLocation doorLocation
+    ) {
+        ClickDoorReturnAction returnAction = door.click(player, itemHeld);
+
+        if (returnAction == ClickDoorReturnAction.CLOSE_DOOR) {
+            door.close();
+            BukkitWorldEditor.closeDoor(doorLocation);
+            return;
+        }
+
+        if (returnAction == ClickDoorReturnAction.OPEN_DOOR) {
+            door.open();
+            BukkitWorldEditor.openDoor(doorLocation);
+            player.removeItem(inventoryIndex);
```

Comment:
Also call player.updateInventory() in order to visually remove the item from the inventory.

Reply:
Thanks! That was exactly what was missing